streams/src/test/java/org/apache/kafka/test/MockClientSupplier.java

```diff
@@ -24,31 +24,32 @@
 import org.apache.kafka.common.serialization.ByteArraySerializer;
 import org.apache.kafka.streams.KafkaClientSupplier;
 
+import java.util.LinkedList;
+import java.util.List;
 import java.util.Map;
 
 public class MockClientSupplier implements KafkaClientSupplier {
     private static final ByteArraySerializer BYTE_ARRAY_SERIALIZER = new ByteArraySerializer();
 
-    public int numberOfCreatedProducers = 0;
-
-    public final MockProducer<byte[], byte[]> producer =
-            new MockProducer<>(true, BYTE_ARRAY_SERIALIZER, BYTE_ARRAY_SERIALIZER);
     public final MockConsumer<byte[], byte[]> consumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
     public final MockConsumer<byte[], byte[]> restoreConsumer = new MockConsumer<>(OffsetResetStrategy.LATEST);
 
+    public final List<Producer> producers = new LinkedList<>();
```

Comment:
We are book-keeping this list but seems never use it: is that intentional? Since we are closing the producers inside Streams anyways I'm wondering what this book-kept list could be used for.